### PR TITLE
Add new implementations for the search results service and the DAO and modify the existing implementations

### PR DESCRIPTION
**Pull request by rahul6603**
Issue - https://github.com/I-TECH-UW/OpenELIS-Global-2/issues/1049

### Summary

- We will create a new implementation of the **_SearchResultsService_** interface in the service layer, called **_LuceneSearchResultsService_** and a new implementation of the **_SearchResultsDAO_** interface, called **_LuceneSearchResultsDAO_**. Currently, these implementations contain placeholders for Apache Lucene search logic which will be added later.
- We will modify the existing service layer implementation **_SearchResultsServiceImpl_** to **_DatabaseSearchResultsService_** and the existing DAO implementation **_SearchResultsDAOImp_** to **_DatabaseSearchResultsDAO_**.
- The `@Primary` annotations on **_DatabaseSearchResultsService_** and **_DatabaseSearchResultsDAO_** are temporary and will be moved to the new implementations once they are fully functional.
- [Reference UML diagram](https://talk.openelis-global.org/t/gsoc-2024-enhanced-search-functionality-for-openelis/731/2?u=rahul6603)

CC: @mozzy11

### CODE REVIEW

**Comment by rahul6603** [comment]
Hi @mozzy11 , could you please review this PR?
Thank you.

File: src/main/java/org/openelisglobal/sample/daoimpl/DatabaseSearchResultsDAO.java [hunk 1]
```diff
@@ -27,30 +27,17 @@
...
 import org.openelisglobal.patientidentitytype.util.PatientIdentityTypeMap;
 import org.openelisglobal.sample.dao.SearchResultsDAO;
+import org.springframework.context.annotation.Primary;
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
 
 @Component
-public class SearchResultsDAOImp implements SearchResultsDAO {
+@Primary
+public class DatabaseSearchResultsDAO implements SearchResultsDAO {
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/sample/daoimpl/DatabaseSearchResultsDAO.java:36]: you can rename this  `DBSearchResultsDAOImp`

File: src/main/java/org/openelisglobal/sample/daoimpl/LuceneSearchResultsDAO.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import javax.transaction.Transactional;
+import org.openelisglobal.common.exception.LIMSRuntimeException;
+import org.openelisglobal.common.provider.query.PatientSearchResults;
+import org.openelisglobal.sample.dao.SearchResultsDAO;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LuceneSearchResultsDAO implements SearchResultsDAO {
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/sample/daoimpl/LuceneSearchResultsDAO.java:12]: rename  to  `LuceneSearchResultsDAOImpl`

File: src/main/java/org/openelisglobal/search/service/DatabaseSearchResultsService.java [hunk 3]
```diff
@@ -4,11 +4,13 @@
...
 import org.openelisglobal.sample.dao.SearchResultsDAO;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 @Service
-public class SearchResultsServiceImpl implements SearchResultsService {
+@Primary
+public class DatabaseSearchResultsService implements SearchResultsService {
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/search/service/DatabaseSearchResultsService.java:13]: rename to `DBSearchResultsServiceImpl`

File: src/main/java/org/openelisglobal/search/service/LuceneSearchResultsService.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.List;
+import javax.transaction.Transactional;
+import org.openelisglobal.common.provider.query.PatientSearchResults;
+import org.openelisglobal.sample.dao.SearchResultsDAO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+@Service
+public class LuceneSearchResultsService implements SearchResultsService {
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/search/service/LuceneSearchResultsService.java:12]: rename to `LuceneResultsServiceImpl`

Comment (rahul6603) [reply src/main/java/org/openelisglobal/search/service/LuceneSearchResultsService.java]: I renamed this service to `LuceneSearchResultsServiceImpl` to stay consistent with the other renamings you suggested. I have also renamed other search service and DAO implementations. Let me know if any other change is needed.
Thank you.

Comment (mozzy11) [reply src/main/java/org/openelisglobal/search/service/LuceneSearchResultsService.java]: Thanks @rahul6603

**Review by mozzy11 — CHANGES_REQUESTED** [review]
@rahul6603 can you add one or two Integration tests for the SearchResultsService for Both DB and Lucene Implemenations and show that currently the DB implementation returns some results and the Lucene Implemenation returns no results

**Pull request merged**